features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioBLE.cpp

```diff
@@ -564,6 +564,9 @@ void BLE::callDispatcher()
         timestamp_t nextTimestamp = (timestamp_t) (WsfTimerNextExpiration(&pTimerRunning) * WSF_MS_PER_TICK) * 1000;
         if (pTimerRunning) {
             nextTimeout.attach_us(timeoutCallback, nextTimestamp);
+        } else {
+            critical_section.disable();
```

Comment:
Why was this, or `CriticalSectionLock critical_section;` needed to begin with?

Reply:
In our case `wsfOsReadyToSleep` and `WsfTimerNextExpiration` needs to be called within a critical section.